interceptors/authentication.go

```diff
@@ -0,0 +1,30 @@
+package interceptors
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/companieshouse/chs.go/log"
+)
+
+func AuthenticationInterceptor(next http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		//Check headers for identity type and identity
+		identityType := r.Header.Get("Eric-Identity-Type")
+		if identityType == "" || identityType != "oauth2" {
```

Comment:
Could probably remove the `identityType == ""` as the second conditional would check for this anyway